docs/create_sla.md

```diff
@@ -27,6 +27,9 @@ def create_sla(self, name, hourly_frequency=None, hourly_retention=None, daily_f
 | archive_name              | str  | The optional archive location you wish to configure on the SLA Domain. When populated, you must also provide a `retention_on_brik_in_days`. |         | None    |
 | retention_on_brik_in_days | int  | The number of days you wish to keep the backups on the Rubrik cluster. When populated, you must also provide a `archive_name`.              |         | None    |
 | instant_archive           | int  | Flag that determines whether or not to enable instant archive. Set to true to enable.                                                       |         | False   |
+| starttime_hour            | int  | (v2 only) Starting hour of allowed backup window. When populated, you must also provide `starttime_min` and `duration_hours`.     |         | None    |
+| starttime_min             | int  | (v2 only) Starting minute of allowed backup window. When populated, you must also provide a `starttime_hour` and `duration_hours`.|         | None    |
+| duration_hours            | int  | (v2 only) Length of allowed backup window. When populated, you must also provide both `startime_min` and `starttime_hour`.                            |         | None    |
```

Comment:
We should match the UI naming conventions as much as possible here. Which in this case means "Snapshot Window".

We can also just have the end user provide us with the full starting time (ex. 1:00 AM) instead of breaking it out. The [_data_time_converstion](https://github.com/rubrikinc/rubrik-sdk-for-python/blob/5255ac4ada8c906dae8a7494efed6788f707353c/rubrik_cdm/data_management.py#L1155 ) function has examples on how to validate this input and then return an error message if it's not formatted correctly. 


`v2` also does not have any meaning in the SDK. Instead, we can describe it has a CDM v5.0 feature.

Reply:
In the API the call is allowedBackupWindow not Snapshot Window. So I think the product itself could use some consistency here. 

I'm ok with stating 5.0 feature but in the SDK tests are all relegated to v1, internal, etc. in name and feature. So again I think we need to decide on some consistency here.

Reply:
Agreed on the consistency between the UI and API. And that is something we are pushing internally. Although, for a lot of reasons that easier said than done. Still, it's something we can keep getting better at.

As a general rule of thumb, our assumption is that our End Users are more familiar with the UI than the API, so we always _try_ to match the language used in the UI.

```
I'm ok with stating 5.0 feature but in the SDK tests are all relegated to v1, internal, etc. in name and feature. So again I think we need to decide on some consistency here.
```

The SDK tests and the main SDK functions have two separate target personas. The SDK is end user facing while the SDK tests are not. The main target audience are developers. That's why it makes sense to mention specific versions in the tests but in the main functions. So the inconsistency here is by design. At the end of the day, the API should be completely transparent to the main users of the SDK.

Reply:
Ok, I see the your point. Do you want me to change the wording to something like:

(CDM 5.0+) Starting minute of Snapshot Window. When populated, you must also provide a `starttime_hour` and `duration_hours`

I'll ping our reps to drive some conversation about the UI <-> API wording. I like the API wording a LOT more than the UI. It's far less confusing to me.

Reply:
> We should match the UI naming conventions as much as possible here. Which in this case means "Snapshot Window".
> 
> We can also just have the end user provide us with the full starting time (ex. 1:00 AM) instead of breaking it out. The [_data_time_converstion](https://github.com/rubrikinc/rubrik-sdk-for-python/blob/5255ac4ada8c906dae8a7494efed6788f707353c/rubrik_cdm/data_management.py#L1155) function has examples on how to validate this input and then return an error message if it's not formatted correctly.
> 
> `v2` also does not have any meaning in the SDK. Instead, we can describe it has a CDM v5.0 feature.

Error message will ensue from the API if time is not formatted correctly. Precheck seems overkill here. Also, the hour/min designation breakout is no different than the UI. You can't put 130am. You have to select 01 hour and 30min and AM. Here we can reduce that burden by using 24 hour clock time making it actually simpler than the UI if only by a single value. Allowing 24h clock and 12hour clock and doing prechecks then doing the conversion to be consumed by the API is quite a lift just so we can say we aren't asking the user for the hour and min.

Reply:
Do you happen to have an example of the error message returned from the API? I'm hoping its something at your finger tips with the code here. If not, no worries I can look it up on my end too.

My main concern is making sure the error messaging is blatantly obvious and often times that means a bit of extra checks in the SDK.

Reply:
Seems pretty clear to me:

![image](https://user-images.githubusercontent.com/29983598/90048097-9e779c00-dc98-11ea-98d4-08f83c094445.png)

Reply:
I've also created features of create_sla() to add replication target, a new update_sla() method, and will be creating an ansible module rubrik_sla that will do complete handling of the SLA as the python-sdk allows instead of being broken into pieces. If you don't have any concerns about the clock I can move forward. I'm holding off on the PRs on that so I can get through not having to conflict fix.